codec/proto_codec.go

```diff
@@ -160,6 +163,35 @@ func (pc *ProtoCodec) MustMarshalJSON(o gogoproto.Message) []byte {
 	return bz
 }
 
+// MarshalAminoJSON provides aminojson.Encoder compatibility with gogoproto messages.
+// x/tx/signing/aminojson cannot marshal gogoproto.message directly since this type is not
+// protoreflect enabled like the std library google.golang.org/protobuf/proto.Message is.
+// So we convert to a dynamicpb message and marshal that directly to JSON.
+func (pc *ProtoCodec) MarshalAminoJSON(msg gogoproto.Message) ([]byte, error) {
+	desc, err := pc.interfaceRegistry.FindDescriptorByName(protoreflect.FullName(gogoproto.MessageName(msg)))
+	if err != nil {
+		return nil, err
+	}
+
+	dynamicMsgType := dynamicpb.NewMessageType(desc.(protoreflect.MessageDescriptor))
```

Comment:
Should we try to find a non-dynamic version in `protoregistry.GlobalTypes` first? That might be slightly more performant than dynamicpb